### PR TITLE
gh-104683: Remove unused variables from `Tools/clinic` and tests for `Tools/clinic`

### PR DESCRIPTION
**Pull request by AlexWaygood**
* Issue: gh-104683

### CODE REVIEW

File: Tools/clinic/cpp.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def _main(filenames: list[str] | None = None) -> None:
             cpp = Monitor(filename, verbose=True)
             print()
             print(filename)
-            for line_number, line in enumerate(f.read().split('\n'), 1):
+            for line in f:
```
Comment (AlexWaygood) [inline Tools/clinic/cpp.py:189]: This line doesn't have any test coverage, but I tested manually by running `python Tools/clinic/cpp.py` on the command line.

File: Lib/test/test_clinic.py [hunk 2]
```diff
@@ -1849,17 +1847,14 @@ def test_scaffolding(self):
...
-            'The igloos are melting!\n'
-        )
         with support.captured_stdout() as stdout:
             errmsg = 'The igloos are melting'
             with self.assertRaisesRegex(clinic.ClinicError, errmsg) as cm:
                 clinic.fail(errmsg, filename='clown.txt', line_number=69)
             exc = cm.exception
             self.assertEqual(exc.filename, 'clown.txt')
             self.assertEqual(exc.lineno, 69)
+            self.assertEqual(stdout.getvalue(), "")
```
Comment (erlend-aasland) [inline Lib/test/test_clinic.py:1857]: So the context manager is not needed either?

**Review by erlend-aasland — APPROVED** [review]

File: Lib/test/test_clinic.py (hunk 2 above)
Comment (AlexWaygood) [reply Lib/test/test_clinic.py:1857]: I guess it's not _strictly_ needed, no. But I think it's maybe useful to assert that `clinic.fail()` never prints anything to stdout?

Comment (erlend-aasland) [reply Lib/test/test_clinic.py:1857]: Yeah, sounds good to me.

**Pull request merged**